src/Microsoft.OData.Core/JsonLight/ODataJsonLightValueSerializer.cs

```diff
@@ -732,6 +736,11 @@ public virtual async Task WriteStreamValueAsync(ODataBinaryStreamValue streamVal
                 stream.Dispose();
                 await streamWriter.EndStreamValueScopeAsync().ConfigureAwait(false);
             }
+
+            if (!streamValue.LeaveOpen)
+            {
+                streamValue.Stream.Dispose();

```

Comment:
Would `Stream.DisposeAsync()` be more appropriate here in case the disposal performs any I/O? (not sure why [line 736](https://github.com/OData/odata.net/pull/2597/files#diff-2cce78e3afeedcc8e1b6c32359a94bdbc867447c0b09b336e2d3c7bcfa6b727eR736) calls `stream.Dispose()`].

Reply:
Aah, good point. Yes, I suggest adding conditional compilation block in that case.

We have currently have code like this in certain areas:

```c#
#if NETCOREAPP3_1_OR_GREATER

#else 

#endif
```
(e.g.: https://github.com/OData/odata.net/pull/2459/files#diff-57ee33c26e730ea9ad1e185c8617b2c707ec2b62e1880df463aab29e668df703R93)

Reply:
We did run into an issue in some edge cases where synchronous disposal lead to synchronous I/O which caused exceptions in AspNetCore because Kestrel disables synchronous I/O by default.

Reply:
Change made. Sadly, I didn't find an efficient way to prevent code duplication (couldn't extract block to its own method due to lack of `Task.CompletedTask` in older runtimes).